SEP-646: Enterprise-Managed Authorization Profile for MCP (#646)
modelcontextprotocol/modelcontextprotocol: docs/specification/draft/basic/enterprise.mdx

---
title: Enterprise Authorization Profile
---

<Info>**Protocol Revision**: draft</Info>

## 1. Introduction

### 1.1 Purpose and Scope

> This document defines a profile of the "[Identity Assertion Authorization Grant](https://datatracker.ietf.org/doc/draft-parecki-oauth-identity-assertion-authz-grant/)" for use within enterprise deployments of the Model Context Protocol (MCP). It specifies how an MCP Client can obtain an access token from an MCP Server's Authorization Server by presenting an identity assertion, and how an MCP Server's Authorization Server validates and utilizes this identity assertion before issuing an access token. This profile is designed to facilitate secure and interoperable authorization within corporate environments, leveraging existing enterprise identity infrastructure.

Author: ```suggestion
This document defines a profile of the "[Identity Assertion Authorization Grant](https://datatracker.ietf.org/doc/draft-parecki-oauth-identity-assertion-authz-grant/)" for use within enterprise deployments of the Model Context Protocol (MCP). It specifies how an MCP Client can obtain an access token from an MCP Server's Authorization Server by presenting an identity assertion, and how an MCP Server's Authorization Server validates and utilizes this identity assertion before issuing an access token. This profile is designed to facilitate secure and interoperable authorization within enterprise environments, leveraging the organization's existing identity infrastructure.
```

Enterprise-ready authorization for MCP Clients and Servers has two key benefits:

> * For end users, this removes the need to manually connect and authorize the MCP Client to individual services within the organization.

Author: ```suggestion
* For end users, this removes the need to manually connect and authorize MCP Clients to individual services within the organization.
```

* For enterprise admins, this enables visibility and control over which MCP Servers are able to be used within the organization.

### 1.2 Roles and Terminology

* **Resource Application**: In the context of this profile, the Resource Application is the MCP Server.

> * **Identity Provider (IdP)**: An entity that creates, maintains, and manages identity information, provides authentication services, and is trusted by a set of applications in an organization's app ecosystem.

Author: It's more just to have a reference for the term within this doc, I think it's fine

* **Identity Assertion**: A security token (e.g., ID Token, SAML Assertion) that contains information about an authenticated user and is issued by an Identity Provider trusted by the Authorization Server.
* **Authorization Server**: This profile uses the term "Authorization Server" to refer to the authorization server that issues tokens to be used at the MCP server.

## 2. Identity Assertion Authorization Grant Overview

This profile uses the "Identity Assertion Authorization Grant" as defined in [draft-parecki-oauth-identity-assertion-authz-grant](https://datatracker.ietf.org/doc/draft-parecki-oauth-identity-assertion-authz-grant/).

The Identity Assertion Authorization Grant follows two steps:

> 1. Token Exchange ([RFC8623](https://datatracker.ietf.org/doc/html/rfc8693))

Author: ```suggestion
1. Single Sign-On to the MCP Client via OpenID Connect or SAML
2. Token Exchange ([RFC8623](https://datatracker.ietf.org/doc/html/rfc8693))
```

> 2. A JWT Authorization Grant ([RFC7523](https://datatracker.ietf.org/doc/html/rfc7523))

Author: ```suggestion
3. A JWT Authorization Grant ([RFC7523](https://datatracker.ietf.org/doc/html/rfc7523))
```

The core flow is as follows:

> * A user logs in to an MCP Client through their enterprise Identity Provider, resulting in an Identity Assertion (ID Token or SAML assertion) being issued to the MCP Client.

Author: There's no requirement in MCP about how the user authenticates, so SAML or OIDC are both fine options. This doesn't affect the OAuth flow of the MCP Client to the MCP Server.

    participant UA as Browser
    participant C as MCP Client
    participant MAS as MCP Authorization Server
    participant MRS as MCP Resource Server
    participant IdP as Identity Provider

    C-->>UA: Redirect to IdP
    UA->>IdP: Redirect to IdP
    Note over IdP: User Logs In
    IdP-->>UA: IdP Authorization Code
    UA->>C: IdP Authorization Code
    C->>IdP: Token Request with IdP Authorization Code
    IdP-->C: ID Token

    note over C: User is logged<br/>in to MCP Client.<br/>Client stores ID Token.

    C->IdP: Exchange ID Token for ID-JAG
    note over IdP: Evaluate Policy
    IdP-->C: Responds with ID-JAG
    C->MAS: Token Request with ID-JAG
    note over MAS: Validate ID-JAG
    MAS-->C: MCP Access Token

    loop
    C->>MRS: Call MCP API with Access Token
    MRS-->>C: MCP Response with Data
    end
```

## 3. User Authentication

To authenticate a user, the MCP Client initiates the process through a request with the IdP using OpenID Connect or SAML.

The following is an example of a web app redirecting the user using OpenID Connect:

```
302 Redirect
Location: https://acme.idp.example/authorize?response_type=code&scope=openid&client_id=...
```

> The user authenticates with the IdP, and is redirected back to the Client with an authorization code, which it can then exchange for an ID Token.

Author: yeah since this starts out as an OIDC flow let's keep it specific

> The enterprise IdP may enforce additional security controls such as multi-factor authentication before granting the user access to the MCP Client.

Author: I think it's important to call out since the target audience for this likely does care about MFA

| `subject_token`        | REQUIRED          | The identity assertion (e.g. the OpenID Connect ID Token or SAML assertion) for the target end-user.                                                                           | (JWT or SAML assertion string)                                         |
| `subject_token_type`   | REQUIRED          | Indicates the type of the security token in the `subject_token` parameter, as specified in [RFC8693 Section 3](https://datatracker.ietf.org/doc/html/rfc8693#section-3).      | `urn:ietf:params:oauth:token-type:id_token` (OIDC)<br/>`urn:ietf:params:oauth:token-type:saml2` (SAML) |

Additional parameters defined in [Section 2.1 of RFC8693](https://datatracker.ietf.org/doc/html/rfc8693#name-request) (`actor_token` and `actor_token_type`) are not used in this specification.

If the IdP requires client authentication when the MCP client performs OpenID Connect for single sign-on, then client authentication of the Token Exchange request is also required.

> The example below illustrates the Token Exchange request, using an ID Token as the Identity Assertion and a client secret as the client authentication method. The ID token is passed as a Subject Token.

Reviewer: ```suggestion
The example below illustrates the Token Exchange request, using an OpenID Connect ID Token as the Identity Assertion and a client secret as the client authentication method. The ID token is passed as a Subject Token.
```

```
POST /oauth2/token HTTP/1.1
Host: acme.idp.example
Content-Type: application/x-www-form-urlencoded

grant_type=urn:ietf:params:oauth:grant-type:token-exchange
&requested_token_type=urn:ietf:params:oauth:token-type:id-jag
&resource=https://acme.chat.example/
&scope=chat.read+chat.history
&subject_token=eyJraWQiOiJzMTZ0cVNtODhwREo4VGZCXzdrSEtQ...
&subject_token_type=urn:ietf:params:oauth:token-type:id_token
&client_id=2ec954a1d60620116d36d9ceb7
&client_secret=a26d84873504215a34a86d52ef5cd64f4b76
```

### 4.1 Processing Rules

> The IdP MUST validate the subject token, and MUST validate that the audience of the Subject Token (e.g. the `aud` claim of the ID Token) matches the `client_id` of the MCP client's client authentication.

Reviewer: What if you split the example out of the explanation? Is this clearer?

```suggestion
The IdP **MUST** validate the Subject Token, and **MUST** validate that the audience of the Subject Token matches the `client_id` of the MCP Client. For example, the IdP could check that the `aud` claim of the ID Token (audience of the Subject Token) matches the `client_id` parameter in the token request (OAuth client authentication for the token request).
```

Agree that we want to make sure this is implemented correctly.

The IdP evaluates administrator-defined policies for the token exchange request and determines if the MCP Client should be granted access to act on behalf of the user for the target MCP Server and scopes.

> The IdP may also introspect the authentication context described in the SSO assertion to determine if step-up authentication is required.

Reviewer: ```suggestion
The IdP may also introspect the authentication context described in the Subject Token to determine if step-up authentication is required.
```

Reviewer: Ah fair. Maybe it's better to use `Identity Assertion` then? I'm trying to use the terminology previously defined and "SSO assertion" was not previously defined as far as I saw.

```
HTTP/1.1 400 Bad Request
Content-Type: application/json
Cache-Control: no-store

{
  "error": "invalid_grant",
  "error_description": "Audience validation failed"
}
```

### 4.3 Identity Assertion Authorization Grant JWT

The Identity Assertion Authorization Grant JWT is issued and signed by the IdP, and describes the intended audience of the authorization grant as well as the client to which it was issued and the subject identifier of the resource owner, using the following claims:

| Parameter    | Required/Optional | Description                                                                                                                    | Reference / Example                                   |
|--------------|-------------------|--------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------|
| `iss`        | REQUIRED          | The IdP issuer URL.                                                                                                            | [Section 4.1.1 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1) |

> | `sub`        | REQUIRED          | The subject identifier (e.g., user ID) of the user at the MCP server.                                                          | [Section 4.1.2 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2) |

Reviewer: ```suggestion
| `sub`        | REQUIRED          | The subject identifier (e.g., user ID) of the user at the MCP Server.                                                          | [Section 4.1.2 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2) |
```

> | `aud`        | REQUIRED          | The Issuer URL of the MCP server's authorization server.                                                                       | [Section 4.1.3 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3) |

Reviewer: ```suggestion
| `aud`        | REQUIRED          | The Issuer URL of the MCP Server's authorization server.                                                                       | [Section 4.1.3 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3) |
```

> | `client_id`  | REQUIRED          | An identifier of the MCP client that this JWT was issued to; SHOULD be a client_id as defined in Section 4.3 of RFC8693.       | [Section 4.3 of RFC8693](https://datatracker.ietf.org/doc/html/rfc8693#section-4.3)     |

Reviewer: ```suggestion
| `client_id`  | REQUIRED          | The identifier of the MCP Client that this JWT was issued to; SHOULD be a client_id as defined in Section 4.3 of RFC8693.       | [Section 4.3 of RFC8693](https://datatracker.ietf.org/doc/html/rfc8693#section-4.3)     |
```

| `scope`      | OPTIONAL          | A JSON string containing a space-separated list of scopes associated with the token, per Section 3.3 of RFC6749.               | `["scope1 scope2"]`<br/>[Section 3.3 of RFC6749](https://datatracker.ietf.org/doc/html/rfc6749#section-3.3)    |

The typ of the JWT indicated in the JWT header MUST be `oauth-id-jag+jwt`.

An example JWT shown with expanded header and payload claims may look like this:

```
{
  "typ": "oauth-id-jag+jwt"
}
.
{
  "jti": "9e43f81b64a33f20116179",
  "iss": "https://acme.idp.example",
  "sub": "U019488227",
  "aud": "https://acme.chat.example/",
  "client_id": "f53f191f9311af35",
  "exp": 1311281970,
  "iat": 1311280970,
  "scope": "chat.read chat.history"
}
.
signature
```

The authorization server **MAY** add additional claims as necessary.

#### Implementation notes

If the IdP is multi-tenant and uses the same issuer for all tenants, the MCP Server will already have IdP-specific logic to determine the tenant from the OpenID Connect ID Token (e.g., the `hd` claim for Google) or SAML assertion, and will need to use that if the IdP also has only one client registration for the MCP Server.

> `sub` should be an opaque ID, as `iss+sub` is unique. The IdP might want to also include the user's email here, which it should do as a new `email` claim. This would let the app dedupe existing users who may have an account with an email address but have not done SSO yet.

Reviewer: > not done SSO yet

I think this could be confusing. Maybe something like:

```suggestion
`sub` should be an opaque ID, as `iss+sub` is unique. The IdP might want to also include additional user information, such as an email address, which it should do as a new `email` claim. This may allow the MCP Client application to properly link existing user accounts to the `sub` identifier used within the enterprise context for SSO.
```

## 5. Access Token Request

The MCP Client makes an access token request to the MCP Server's authorization server using the previously obtained Identity Assertion Authorization Grant as a JWT Assertion as defined by [RFC7523](https://datatracker.ietf.org/doc/html/rfc7523).

| Parameter   | Required/Optional | Description                                                                                                        | Example/Allowed Values                                                                           |
|-------------|-------------------|--------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|

> | `grant_type`| REQUIRED          | MUST be `urn:ietf:params:oauth:grant-type:jwt-bearer`                                                              | `urn:ietf:params:oauth:grant-type:jwt-bearer`<br/>[Section 4.1 of RFC7523](https://datatracker.ietf.org/doc/html/rfc7523#section-4.1) |

Reviewer: ```suggestion
| `grant_type`| REQUIRED          | **MUST** be `urn:ietf:params:oauth:grant-type:jwt-bearer`                                                              | `urn:ietf:params:oauth:grant-type:jwt-bearer`<br/>[Section 4.1 of RFC7523](https://datatracker.ietf.org/doc/html/rfc7523#section-4.1) |
```

| `assertion` | REQUIRED          | The Identity Assertion Authorization Grant JWT obtained in the previous token exchange step                        | (JWT string)                                                                                     |

The MCP Client authenticates with its credentials as registered with the MCP Server's authorization server.

> If the MCP Client has not yet registered with the MCP Server's authorization server, then it can do a Dynamic Client Registration request at this stage.

Reviewer: link to DCR and/or part of the Authorization spec that discusses this?

An example request may look like this:

```
POST /oauth2/token HTTP/1.1
Host: acme.chat.example
Authorization: Basic yZS1yYW5kb20tc2VjcmV0v3JOkF0XG5Qx2

grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer
assertion=eyJhbGciOiJIUzI1NiIsI...
```

### 5.1 Processing Rules

All of [Section 5.2 of RFC7521](https://datatracker.ietf.org/doc/html/rfc7521#section-5.2) applies, in addition to the following processing rules:

* Validate the JWT `typ` is `oauth-id-jag+jwt` (per [RFC8725](https://datatracker.ietf.org/doc/html/rfc8725))

> * The `aud` claim **MUST** identify the Issuer URL of the MCP Server's authorization server as the intended audience of the JWT.
> * The `client_id` claim **MUST** identify the same client as the client authentication in the request.

Reviewer: I think we need something to ground the last sentence about the `client_id` in the real world. "client authentication" is clear to an expert, but not obvious what it means to a casual implementer IMO.

```suggestion
* The `aud` claim of the assertion JWT **MUST** identify the Issuer URL of the MCP Server's authorization server as the intended audience of the JWT.
* The `client_id` claim of the assertion JWT **MUST** identify the same client as the client authentication in the request (e.g., the `client_id` specified in a basic auth header).
```

Author: http basic auth is sadly still far more common than private key JWT for OIDC integrations, so I think this is the more helpful example in this case.

This provides two key benefits:

* For end users, this removes the need to manually connect and authorize the MCP Client to individual services within the organization.
* For enterprise admins, this enables visibility and control over which MCP Servers are able to be used within the organization.

### 1.2 Roles and Terminology

* **Resource Application**: In the context of this profile, the Resource Application is the MCP Server.
* **Identity Provider (IdP)**: An entity that creates, maintains, and manages identity information for principals and provides authentication services, and is trusted by a set of applications in an organization's app ecosystem.
* **Identity Assertion**: A security token (e.g., ID Token, SAML Assertion) that contains information about an authenticated user and is issued by an Identity Provider trusted by the Authorization Server.
* **Authorization Server**: This profile uses the term "Authorization Server" to refer to the authorization server that issues tokens to be used at the MCP server.

> 

Author: Technically the ID token doesn't represent the user, it represents the authentication event of the user. I added slightly different definitions of these.

* The MCP Client uses the access token to make requests to the MCP Server.

### 2.1 Sequence Diagram

The following diagram outlines an example flow:

```mermaid
sequenceDiagram
    participant UA as Browser
    participant C as MCP Client
    participant MAS as MCP Authorization Server
    participant MRS as MCP Resource Server
    participant IdP as Identity Provider

    C-->>UA: Redirect to IdP
    UA->>IdP: Redirect to IdP
    Note over IdP: User Logs In
    IdP-->>UA: IdP Authorization Code
    UA->>C: IdP Authorization Code
    C->>IdP: Token Request with IdP Authorization Code
    IdP-->C: ID Token

    note over C: User is logged<br>in to MCP Client.<br>Client stores ID Token.

    C->IdP: Exchange ID Token for ID-JAG
    note over IdP: Evaluate Policy
    IdP-->C: Responds with ID-JAG
    C->MAS: Token Request with ID-JAG
    note over MAS: Validate ID-JAG
    MAS-->C: MCP Access Token

    loop
    C->>MRS: Call MCP API with Access Token
    MRS-->>C: MCP Response with Data
    end
```

## 3. User Authentication

The Client initiates an authentication request with the IdP using OpenID Connect or SAML.

> The following is an example of a web app redirecting the user using OpenID Connect:

Author: This is meant to show the diagram version of the flow described above, I'll see if I can clear that up

}
```

### 4.3 Identity Assertion Authorization Grant JWT

The Identity Assertion Authorization Grant JWT is issued and signed by the IdP, and describes the intended audience of the authorization grant as well as the client to which it was issued and the subject identifier of the resource owner, using the following claims:

`iss`: REQUIRED - The IdP issuer URL as defined in Section 4.1.1 of [RFC7519]

`sub`: REQUIRED - The subject identifier (e.g. user ID) of the user at the MCP server as defined in Section 4.1.2 of [RFC7519]

`aud`: REQUIRED - The Issuer URL of the MCP server's authorization server as defined in Section 4.1.3 of [RFC7519]

`client_id`: REQUIRED - An identifier of the MCP client that this JWT was issued to. For interoperability, the client identifier SHOULD be a client_id as defined in Section 4.3 [RFC8693].

`jti`: REQUIRED - Unique ID of this JWT as defined in Section 4.1.7 of [RFC7519]

`exp`: REQUIRED - as defined in Section 4.1.4 of [RFC7519]

`iat`: REQUIRED - as defined in Section 4.1.6 of [RFC7519]

`scope`: OPTIONAL - a JSON string containing a space-separated list of scopes associated with the token, in the format described in Section 3.3 of [RFC6749].

> The typ of the JWT indicated in the JWT header MUST be `oauth-id-jag+jwt`.

Author: This actually should be `typ` with backticks, that's the actual JWT claim name in the header.

> The typ of the JWT indicated in the JWT header MUST be `oauth-id-jag+jwt`.

Reviewer: Along those lines:

```suggestion
The `typ` claim of the JWT indicated in the JWT header **MUST** be `oauth-id-jag+jwt`.
```

### 1.2 Roles and Terminology

- **Resource Application**: In the context of this profile, the Resource Application is the MCP Server.
- **Identity Provider (IdP)**: An entity that creates, maintains, and manages identity information, provides authentication services, and is trusted by a set of applications in an organization's app ecosystem.
- **Identity Assertion**: A security token (e.g., ID Token, SAML Assertion) that contains information about an authenticated user and is issued by an Identity Provider trusted by the Authorization Server.
- **Authorization Server**: This profile uses the term "Authorization Server" to refer to the authorization server that issues tokens to be used at the MCP server.
- **Client**: In the context of this profile, the Client is the MCP Client.
- **ID Token**: A security token that contains claims about the authentication of a user when using a client, as defined in [OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html#IDToken).
- **Subject Token**: A security token that represents the identity of the user, such as an ID Token.

> 

Reviewer 2: Should we add a definition here was was JAG stands for (JWT Authorization Grant)?

1. Single Sign-On to the MCP Client via OpenID Connect or SAML
2. Token Exchange ([RFC8623](https://datatracker.ietf.org/doc/html/rfc8693))
3. A JWT Authorization Grant ([RFC7523](https://datatracker.ietf.org/doc/html/rfc7523))

The core flow is as follows:

- A user logs in to an MCP Client through their enterprise Identity Provider, resulting in an Identity Assertion (ID Token or SAML assertion) being issued to the MCP Client.
- The MCP Client sends a Token Exchange [[RFC8623](https://datatracker.ietf.org/doc/html/rfc8693)] request to the Identity Provider including the identity assertion and identifier of the MCP Server it is attempting to access, and obtains a Identity Assertion Authorization Grant.
- The MCP Client uses the Identity Assertion Authorization Grant as a JWT Authorization Grant [[RFC7523](https://datatracker.ietf.org/doc/html/rfc7523)] to request an access token from the Authorization Server.
- The Authorization Server validates the Identity Assertion Authorization Grant and, if valid, issues an access token.
- The MCP Client uses the access token to make requests to the MCP Server.

### 2.1 Sequence Diagram

> The following diagram outlines an example flow:

Reviewer 2: It would be helpful to set the context here with the example scenario where SSO is enabled, and it allows the same identity issued by the AS (Okta) protecting the MCP server to access the downstream resource (Slack) outside the "domain" using this flow. Like the Slack example in your blog post. I'm not entirely sure how as the example in the spec needs to be general enough to capture all scenario but maybe there is room for a example use cases section?

Reviewer 2: Yes between the first paragraph and benefits would be an ideal place. It will really make the following sections more impactful.

The example below illustrates the Token Exchange request, using an OpenID Connect ID Token as the Identity Assertion and a client secret as the client authentication method. The ID token is passed as a Subject Token.

```
POST /oauth2/token HTTP/1.1
Host: acme.idp.example
Content-Type: application/x-www-form-urlencoded

grant_type=urn:ietf:params:oauth:grant-type:token-exchange
&requested_token_type=urn:ietf:params:oauth:token-type:id-jag
&audience=https://auth.chat.example/
&resource=https://mcp.chat.example/
&scope=chat.read+chat.history
&subject_token=eyJraWQiOiJzMTZ0cVNtODhwREo4VGZCXzdrSEtQ...
&subject_token_type=urn:ietf:params:oauth:token-type:id_token
&client_id=2ec954a1d60620116d36d9ceb7
&client_secret=a26d84873504215a34a86d52ef5cd64f4b76
```

### 4.1 Processing Rules

The IdP MUST validate the Subject Token, and MUST validate that the audience of the Subject Token (e.g. the `aud` claim of the ID Token) matches the Client Identifier of the MCP client making this request (e.g. by checking the `client_id` in the client authentication).

> The IdP evaluates administrator-defined policies for the token exchange request and determines if the MCP Client should be granted access to act on behalf of the user for the target MCP Server and scopes.

Author: I added a section at the bottom for this: https://mcp-staging-aaronpk-enterprise-mcp.mintlify.app/specification/draft/basic/enterprise_managed_authorization#6-2-scope-of-enterprise-visibility-and-policy-enforcement

The IdP may also introspect the authentication context described in the Identity Assertion to determine if step-up authentication is required.

### 4.2 Token Exchange Response

If access is granted, the IdP creates a signed Identity Assertion Authorization Grant JWT and returns it in the token exchange response defined in [Section 2.2 of RFC8693](https://datatracker.ietf.org/doc/html/rfc8693#section-2.2):

```
HTTP/1.1 200 OK
Content-Type: application/json
Cache-Control: no-store
Pragma: no-cache

{
  "issued_token_type": "urn:ietf:params:oauth:token-type:id-jag",

>   "access_token": "eyJhbGciOiJIUzI1NiIsI...",

Reviewer 2: @desimone am I missing something or the "RE: Bearer Token Exposure" risk related to "[LLM-based agents might not be the best at keeping secrets](https://simonwillison.net/series/prompt-injection/)" only applies if the LLM sees the token? Yes in this case the agent (MCP client) has the token but it doesn't mean it gets sent in a prompt to the LLM right?

>   "access_token": "eyJhbGciOiJIUzI1NiIsI...",

Author: This is also not a new problem introduced by this profile, these risks also exist in the standard interactive OAuth model.

| `iss`        | REQUIRED          | The IdP issuer URL.                                                                                                            | [Section 4.1.1 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1) |
| `sub`        | REQUIRED          | The subject identifier (e.g., user ID) of the user at the MCP server.                                                          | [Section 4.1.2 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2) |
| `aud`        | REQUIRED          | The Issuer URL of the MCP server's authorization server.                                                                       | [Section 4.1.3 of RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3) |

> | `client_id`  | REQUIRED          | An identifier of the MCP client registered at the Authorization Server that this JWT was issued to; SHOULD be a client_id as defined in Section 4.3 of RFC8693.       | [Section 4.3 of RFC8693](https://datatracker.ietf.org/doc/html/rfc8693#section-4.3)     |

Reviewer: ```suggestion
| `client_id`  | REQUIRED          | An identifier of the MCP Client registered at the Authorization Server that this JWT was issued to; SHOULD be a client_id as defined in Section 4.3 of RFC8693.       | [Section 4.3 of RFC8693](https://datatracker.ietf.org/doc/html/rfc8693#section-4.3)     |
```
